lib/attr_extras/attr_initialize.rb

```diff
@@ -12,22 +12,13 @@ def apply
     names = @names
     block = @block
     validate_arity = method(:validate_arity)
-    set_ivar_from_hash = method(:set_ivar_from_hash)
+    set_ivar_from_args = method(:set_ivar_from_args)
 
     klass.send(:define_method, :initialize) do |*values|
       validate_arity.call(values.length, self.class)
 
       names.zip(values).each do |name_or_names, value|
```

Comment:
Also, if I change the test to do `example = klass.new("Foo", "Rab", "C", :baz => "Zab")` it tells me "wrong number of arguments (4 for 1..3)". But 1 required positional + 2 optional positional + one optional hash should be "1..4", right?